api/censusdb/censusdb.go

```diff
@@ -0,0 +1,199 @@
+package censusdb
+
+import (
+	"bytes"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/google/uuid"
+	"go.vocdoni.io/dvote/censustree"
+	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+const (
+	censusDBprefix          = "cs_"
+	censusDBreferencePrefix = "cr_"
+)
+
+// CensusRef is a reference to a census.
+type CensusRef struct {
+	tree       *censustree.Tree // must be private to avoid gob serialization
+	AuthToken  *uuid.UUID
+	CensusType int32
+	Indexed    bool
+	IsPublic   bool
+}
+
+// Tree returns the censustree.Tree object of the census reference.
+func (cr *CensusRef) Tree() *censustree.Tree {
+	return cr.tree
+}
+
+// SetTree sets the censustree.Tree object to the census reference.
+func (cr *CensusRef) SetTree(tree *censustree.Tree) {
+	cr.tree = tree
+}
```

Comment:
just make the tree field public :) to avoid encoding it in the gob, you could do something like:

```
type CensusRef struct {
    Tree ...
    Encoded EncodedCensusRef
}

type EncodedCensusRef struct {
    AuthToken ...
    ...
}
```

Then you only encode and decode the `Encoded` field. No real difference in terms of memory, but you do avoid the get/set methods.

Reply:
Nice trick. I was looking for a struct tag able to exclude a value from gob but it does not exist. However I like more the get/set approach since we are only avoiding 1 field from serialization (fore more, I'd see your approach better).